storage/namespace.go

```diff
@@ -406,18 +407,24 @@ func (n *dbNamespace) Tick(c context.Cancellable) {
 			if c.IsCancelled() {
 				return
 			}
-			shardResult := shard.Tick(c)
+
+			shardResult, err := shard.Tick(c)
 
 			l.Lock()
 			r = r.merge(shardResult)
+			multiErr = multiErr.Add(err)
 			l.Unlock()
 		})
 	}
 
 	wg.Wait()
 
 	if c.IsCancelled() {
-		return
+		return multiErr.FinalError()
```

Comment:
Is it worth actually returning here?  As long as err == nil even if it just got cancelled we should probably inc stats, etc.